### PR TITLE
[issue-687] Refactor/operator register api

### PR DESCRIPTION
**Pull request by heesuk-ahn**
## Proposal

* Change deprecated kubernetesClient.getVersion to KubernetesClient.getKubernetesVersion
* Pass a non-null value as a parameter by creating defaultConfiguration in register(Reconciler reconciler)
* Remove unnecessary null defense code

by #687

### CODE REVIEW

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Operator.java [hunk 1]
```diff
@@ -127,39 +128,34 @@ public void close() {
...
-          configuration.getCustomResourceClass(),
-          watchedNS);
     }
+
+    final var controller = new Controller<>(reconciler, configuration, kubernetesClient);
+
+    controllers.add(controller);
+
+    final var watchedNS =
+        configuration.watchAllNamespaces() ? "[all namespaces]" : configuration.getEffectiveNamespaces();
```
Comment (csviri) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Operator.java:152]: This line does not seems to be formatted if you just build the project with maven it should fix it.

**Comment by csviri** [comment]
Just made a minor not, otherwise LGTM!! Thank you!

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Operator.java (hunk 1 above)
Comment (heesuk-ahn) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Operator.java]: Thanks for the reply :) 
 I rebuild maven and made a commit by modifying the code formats.

**Review by csviri — APPROVED** [review]

File: operator-framework-core/src/test/java/io/javaoperatorsdk/operator/OperatorTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    when(configurationService.getConfigurationFor(fooReconciler)).thenReturn(configuration);
+    when(configuration.watchAllNamespaces()).thenReturn(true);
+    when(configuration.getName()).thenReturn("FOO");
+    when(configuration.getCustomResourceClass()).thenReturn(FooReconciler.class);
+
+    // when
+    operator.register(fooReconciler);
+
+    // then
+    verify(configuration).watchAllNamespaces();
```
Comment (metacosm) [inline operator-framework-core/src/test/java/io/javaoperatorsdk/operator/OperatorTest.java:41]: Not sure I like this test as this is more testing the internal implementation than checking that the reconciler is actually registered.

**Review by metacosm — CHANGES_REQUESTED** [review]

File: operator-framework-core/src/test/java/io/javaoperatorsdk/operator/OperatorTest.java (hunk 2 above)
Comment (heesuk-ahn) [reply operator-framework-core/src/test/java/io/javaoperatorsdk/operator/OperatorTest.java:42]: @metacosm Thanks for your advice!  I added a test code to check if the controller was added normally. :) 

```
    assertThat(operator.getControllers().size()).isEqualTo(1);
    assertThat(operator.getControllers().get(0).getReconciler()).isEqualTo(fooReconciler);
```

**Comment by heesuk-ahn** [comment]
As #678 was merged into the base branch `v2`, there was a conflict with some codes of the PR. I'll fix this part and rebase it. :)

**Comment by heesuk-ahn** [comment]
> As #678 was merged into the base branch `v2`, there was a conflict with some codes of the PR. I'll fix this part and rebase it. :)

**Comment by heesuk-ahn** [comment]
> As #678 was merged into the base branch `v2`, there was a conflict with some codes of the PR. I'll fix this part and rebase it. :)

Conflict has been corrected and rebase has been completed with v2. :)

**Review by metacosm — APPROVED** [review]

**Comment by csviri** [comment]
thank you @heesuk-ahn !

**Pull request merged**